wallet/src/account/mod.rs

```diff
@@ -868,5 +934,50 @@ fn group_utxos_for_input<T, Grouped: Clone>(
     Ok(tokens_grouped)
 }
 
+pub fn tx_size_with_outputs(outputs: &[TxOutput]) -> usize {
+    let tx = SignedTransaction::new(
+        Transaction::new(1, vec![], outputs.into()).expect("should not fail"),
+        vec![],
+    )
+    .expect("should not fail");
+    serialization::Encode::encoded_size(&tx)
+}
+
+fn input_signature_size(destination: &Destination) -> usize {
+    // Sizes calculated upfront
+    match destination {
+        Destination::Address(_) => 103,
+        Destination::PublicKey(_) => 69,
+        Destination::AnyoneCanSpend => 2,
+        Destination::ScriptHash(_) | Destination::ClassicMultisig(_) => unimplemented!(),
```

Comment:
This is a bad idea. Please return an error instead. We don't want transactions to cause crashes in the wallet.